### PR TITLE
68 documents catalogue page

### PR DESCRIPTION
**Pull request by fructusmortus**

### CODE REVIEW

File: module-3/back-end/index.js [hunk 1]
```diff
@@ -43,6 +42,16 @@ app.get('/selectAllDocuments', async (req, res) => {
   }
 })
 
+app.get('/selectAllDocuments/:projectId', async (req, res) => {
+  try {
+    const result = await Document.selectAllDocumentsByProject(req.params.projectId)
```
Comment (shelby-carter) [inline module-3/back-end/index.js:47]: I think `selectAllDocumentsByProjectId` would be a better name for this method

File: module-3/docker-compose.yml [hunk 2]
```diff
@@ -30,14 +30,14 @@ services:
...
+  #   image: node:13
+  #   volumes:
+  #     - ./web-app:/usr/src/service
+  #   working_dir: /usr/src/service
+  #   command: npm run start
+  #   ports:
+  #     - 80:3000
+  #   depends_on:
+  #     - "db"
+  #     - "back-end"
```
Comment (shelby-carter) [inline module-3/docker-compose.yml:43]: NEW LINE!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

File: module-3/web-app/src/components/Pages/Documents/Documents.jsx [hunk 3]
```diff
@@ -1,10 +1,30 @@
...
+import {useSearchParams} from "react-router-dom"
+import style from "./Documents.module.scss"
 
+//TODO in progress documents catalogue page
 const Documents = () => {
+    const [documents, setDocuments] = useState('')
+    const [searchParams, setSearchParams] = useSearchParams()
+    const projectId = searchParams.get('projectid')
+   
+    useEffect(() => {
```
Comment (shelby-carter) [inline module-3/web-app/src/components/Pages/Documents/Documents.jsx:11]: Add `TODO` comment to move this in component fetch to a redux action later on

File: module-3/web-app/src/components/Pages/Projects/Projects.jsx [hunk 4]
```diff
@@ -1,30 +1,35 @@
...
     useEffect(() => {
-        async function getData(url = "http://localhost:3001/selectAllProjects") {
-            const response = await fetch(url, {
-                method: 'GET'
-            })
-
-            return response.json() 
-        }
-        getData().then(res => setProjects(res))
+        fetch("http://localhost:3001/selectAllProjects/")
```
Comment (shelby-carter) [inline module-3/web-app/src/components/Pages/Projects/Projects.jsx:11]: Add TODO comment to move this in component fetch to a redux action later on

File: module-3/web-app/src/components/Pages/Projects/Projects.jsx [hunk 5]
```diff
@@ -1,30 +1,35 @@
...
+            .then(res => res.json())
+            .then(data => setProjects(data))
     }, [])
+
```
Comment (shelby-carter) [inline module-3/web-app/src/components/Pages/Projects/Projects.jsx:15]: Please clean up commented code

File: module-3/back-end/index.js (hunk 1 above)
Comment (fructusmortus) [reply module-3/back-end/index.js:47]: Make sense

File: module-3/back-end/src/models/document.js [hunk 6]
```diff
@@ -30,16 +62,36 @@ const Model = (config) => {
...
+   
+    const selectAllDocumentsByProject = async projectId => {
+        const documents = await knex.select(
+            'document.id as id',
+            'document.title as title',
+            'document.content as content',
+            'document.author_id as authorId',
+            'document.status_id as statusId',
+            'document.updated_at as updatedAt',
+            'project_documents.document_id as documentId',
```
[the remark this answers is not in the document]
Comment (fructusmortus) [reply module-3/back-end/src/models/document.js:77]: Yes, thanks!

File: module-3/docker-compose.yml (hunk 2 above)
Comment (fructusmortus) [reply module-3/docker-compose.yml:43]: Done

File: module-3/web-app/src/components/Pages/Documents/Documents.jsx (hunk 3 above)
Comment (fructusmortus) [reply module-3/web-app/src/components/Pages/Documents/Documents.jsx:11]: Done

File: module-3/web-app/src/components/Pages/Projects/Projects.jsx (hunk 4 above)
Comment (fructusmortus) [reply module-3/web-app/src/components/Pages/Projects/Projects.jsx:11]: Done

File: module-3/web-app/src/components/Pages/Projects/Projects.jsx (hunk 5 above)
Comment (fructusmortus) [reply module-3/web-app/src/components/Pages/Projects/Projects.jsx:15]: Done

**Pull request merged**